tests/e2e-pw/utils/merchant.ts

```diff
@@ -90,26 +90,33 @@ export const deactivateMulticurrency = async ( page: Page ) => {
 	await saveWooPaymentsSettings( page );
 };
 
-export const addMulticurrencyWidget = async ( page: Page ) => {
+export const addMulticurrencyWidget = async (
+	page: Page,
+	blocksVersion = false
+) => {
 	await navigation.goToWidgets( page );
 	// Wait for all widgets to load. This is important to prevent flakiness.
-	await page.locator( '.components-spinner' ).first().waitFor();
 	await expect( page.locator( '.components-spinner' ) ).toHaveCount( 0 );
 
 	if ( await page.getByRole( 'button', { name: 'Close' } ).isVisible() ) {
 		await page.getByRole( 'button', { name: 'Close' } ).click();
 	}
 
-	const isWidgetAdded = await page
-		.locator( 'iframe[srcdoc*=currency]' )
-		.first()
-		.isVisible();
+	// At this point, widgets might still be loading individually.
+	await expect( page.locator( '.components-spinner' ) ).toHaveCount( 0 );
```

Comment:
@ismaeldcom, I noticed you changed this today via #10204 and we came to different approaches for the same issue I guess.

In this approach I added a second check for the loading spinners because I realized the widget page has two loading moments: 1st is when the page is loading and 2nd is when all widgets have been loaded but some start to load their own content individually.

I don't have a strong preference for either of the approaches, but feel free to raise any concerns you may have with this change on top of yours.

Reply:
Yes, I noticed that we were expecting the spinners to be zero before they were added, that's why I added that `waitFor`. No preference from my side, if it works fine, feel free to update it 🙂